### PR TITLE
ATO-322: Add alerting for lambdas

### PR DESCRIPTION
**Pull request by hannah-leong**
## What

Add alerting for lambdas to the cloudformation template; done for OpenIdConfigurationFunction - will be copied and pasted for others as they are moved across

## 🚨⚠️ Orchestration and Authentication mutual dependencies ⚠️ 🚨

Be careful when making changes to code in 'shared' components where each team has a copy.
Check with counterparts to see if changes need to be made in the other team's code.

In particular pay attention to classes representing Session data where changes need to be applied on both sides to avoid deserialization errors.

- [ ] Impact on orch and auth mutual dependencies has been checked.

### CODE REVIEW

**Comment by hannah-leong** [comment]
Note: currently using `AWS::AccountId` rather than the alias as there seems to be no easy way of doing it. Will look into this on Monday

File: template.yaml [hunk 1]
```diff
@@ -80,10 +80,39 @@ Globals:
         - SecretArn: !FindInMap [ EnvironmentConfiguration, !Ref Environment, dynatraceSecretArn ]
 
 Resources:
+  MainKmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: Key used to encrypt the Orchestration
```
Comment (ethanmills) [inline template.yaml:86]: this doesn't sound quite right

Comment (hannah-leong) [reply template.yaml:86]: :(

File: template.yaml [hunk 2]
```diff
@@ -110,4 +139,85 @@ Resources:
...
+              MetricName: Errors
+              Dimensions:
+                - Name: FunctionName
+                  Value: !Ref OpenIdConfigurationFunction.Alias
+            Period: 60
+            Stat: Sum
+            Unit: Count
+
+
+  OpenIdConfigurationFunctionSlackEvents:
```
Comment (ethanmills) [inline template.yaml:219]: I think this resource name is wrong. We're only going to have one of these, not one per lambda right?

Comment (hannah-leong) [reply template.yaml:219]: Ah yes, good point, shall change that now

**Comment by ethanmills** [comment]
> Note: currently using `AWS::AccountId` rather than the alias as there seems to be no easy way of doing it. Will look into this on Monday

I'd be tempted to define a mapping between account number and name if there's no easier way. I think it's quite important that we have something human readable, as that's what shows up in the text of the alert and that's where TSD start from

**Review by ethanmills — APPROVED** [review]
Discussed we still need to figure out how to get these into slack

**Pull request merged**